build.py

```diff
@@ -804,14 +807,14 @@ def write(self):
         for default in sorted(strmap(DEFAULT_TARGETS)):
             ninja.default(default)
 
-        # Tell vim not to break up long lines.
+        # Tell vim and emacs not to break up long lines.
         ninja.newline()
         ninja.comment('vim: set textwidth=0 :')
-
-        with open(self.ninja_file, 'w') as file:
-            file.write(content.getvalue())
+        ninja.comment('-*- eval: (auto-fill-mode -1) -*-')
+        with open(self.ninja_file, 'wb') as f:
```

Comment:
Blarg. github won't let me comment on like 791, so I'll put this here. Try changing from `BytesIO` to `StringIO`. That may avoid some of the encoding changes you are doing.

Reply:
Done

Reply:
Does this need to be `'wb'` rather than just `'w'` now that we are using `io.BytesIO`?